File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.js
function solution(keyinput, board) {
  var answer = [];
  let initX = 0;
  let initY = 0;
  const boundaryValueX = parseInt(board[0] / 2)
  const boundaryValueY = parseInt(board[1] / 2)
  keyinput.forEach((key) => {
    if (key === 'left' && initX !== -boundaryValueX) {
      initX -= 1;
    } else if (key === 'right' && initX !== boundaryValueX) {
      initX += 1;
    } else if (key === 'up' && initY !== boundaryValueY) {
      initY += 1;
    } else if (key === 'down' && initY !== -boundaryValueY) {
      initY -= 1;
    }
  })
  answer.push(initX);
  answer.push(initY);
  return answer;
}

File: 프로그래머스/lv0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    answer = [[0, 0]]
    boundaryX = board[0] // 2
    boundaryY = board[1] // 2
    
    for cmd in keyinput:
        x, y = answer.pop()
        if cmd == "up":
            nx, ny = x, y + 1
        elif cmd == "down":
            nx, ny = x, y - 1
        elif cmd == "left":
            nx, ny = x - 1, y
        else:
            nx, ny = x + 1, y
        if -boundaryX <= nx <= boundaryX and -boundaryY <= ny <= boundaryY:
            answer.append([nx, ny])
        else:
            answer.append([x, y])
        print(answer, boundaryX)
    return answer[0]